vmdb/lib/workers/mixins/event_catcher_openstack_mixin.rb

```diff
@@ -0,0 +1,74 @@
+module EventCatcherOpenstackMixin
+  # seems like most of this class could be boilerplate when compared against EventCatcherRhevm
+  def event_monitor_handle
+    require 'openstack/openstack_event_monitor'
+    unless @event_monitor_handle
+      options = {}
+      options[:hostname] = @ems.ipaddress
+      options[:port]     = self.worker_settings[:amqp_port]
+      if @ems.has_authentication_type? :amqp
+        options[:username] = @ems.authentication_userid(:amqp)
+        options[:password] = @ems.authentication_password(:amqp)
+      end
+      options[:topics] = self.worker_settings[:topics]
+      options[:duration] = self.worker_settings[:duration]
+      options[:capacity] = self.worker_settings[:capacity]
+      @event_monitor_handle = OpenstackEventMonitor.new(options)
+    end
+    @event_monitor_handle
+  end
+
+  def reset_event_monitor_handle
+    @event_monitor_handle = nil
+  end
+
+  def stop_event_monitor
+    begin
+      @event_monitor_handle.stop unless @event_monitor_handle.nil?
+    rescue Exception => err
+      $log.warn("#{self.log_prefix} Event Monitor Stop errored because [#{err.message}]")
+      $log.warn("#{self.log_prefix} Error details: [#{err.details}]")
+      $log.log_backtrace(err)
+    ensure
+      reset_event_monitor_handle
+    end
+  end
+
+  def monitor_events
+    begin
+      event_monitor_handle.start
+      event_monitor_handle.each_batch do |events|
+        $log.debug("#{self.log_prefix} Received events #{events.collect {|e| e.payload["event_type"]}}") if $log.debug?
```

Comment:
@blomquisg damn, have you checked also overcloud? If they are the same?

Nevertheless we can just merge this patch, it at least removes the failing worker.